carto_report/report.py

```diff
@@ -24,614 +25,595 @@
         
-            'today': today,
+        real_storage = quota*2
+        used_storage = round(dsets_size,2)
+        pc_used = round(used_storage*100.00/real_storage,2)
+        storage = [real_storage, 'carto', 'storage', 'false', used_storage, pc_used]
+        lds.loc[len(lds)] = storage
         
-            'real_storage':real_storage,
-            'used_storage':used_storage,
-            'pc_used':pc_used,
-            'left_storage':left_storage,
-            'pc_left':pc_left,
+        lds = lds.set_index('Service')
+        lds['Left'] = round(lds['Monthly Quota'] - lds['Used'],1)
+        lds['% Left'] = 100.00 - lds['% Used']
+        lds_df = lds[['Monthly Quota', 'Provider', 'Used', '% Used', 'Left', '% Left']]
+
+        return lds_df
+
+    ### get analysis and tables data
+
+    def getSizes(self, dsets_df):
+        '''
+        Method to get all tables sizes, know cartodbfied and non cartodbfied tables (analysis).
+        '''
+        
+        self.logger.info('Getting list of tables...')
         
-            'total_maps':total_maps,
-            'total_analysis':total_analysis,
-            'total_size_analysis':total_size_analysis,
-            'analysis_df': analysis_df,
-            'top_5_maps_date': top_5_maps_date,
+        all_tables = self.sql.send(
+                    "select pg_class.relname as name from pg_class, pg_roles, pg_namespace" +
+                    " where pg_roles.oid = pg_class.relowner and " +
+                    "pg_roles.rolname = current_user " +
+                    "and pg_namespace.oid = pg_class.relnamespace and pg_class.relkind = 'r'")['rows']
+
+        all_tables_df = json_normalize(all_tables)
         
-            'sync': sync,
-            'total_dsets':total_dsets,
-            'total_size_tbls':total_size_tbls,
-            'private':private,
-            'link':link,
-            'public':public,
-            'geo':geo,
-            'points':points,
-            'lines':lines,
-            'polys':polys,
-            'none_tbls':none_tbls,
-            'top_5_dsets_size': top_5_dsets_size,
-            'top_5_dsets_date': top_5_dsets_date,
         
-            'lds': lds,
+        self.logger.info('Retrieved {} tables.'.format(len(all_tables_df)))
         
-            'html_fig1': fig_to_html(fig1),
-            'html_fig2': fig_to_html(fig2)
-        })
+        dsets_df['cartodbfied'] = 'Yes'
+        all_tables_df = all_tables_df.merge(dsets_df, on='name', how='left')
+        all_tables_df['cartodbfied'] = all_tables_df['cartodbfied'].fillna('No')
+        all_tables_df['size'] = 0
+        
+        self.logger.info('Getting table sizes...')
+        
+        for index, row in all_tables_df.iterrows():
+            try:
+                size = self.sql.send("select pg_total_relation_size('" + row['name'] + "') as size")['rows'][0].get('size')
+            except:
+                self.logger.info('Error at: ' + str(row['name']))
```

Comment:
This should be logged as an error or a warning at least

Reply:
What is the proper way to do this? Something like...

```python
except Exception as ex:
  self.logger.info('Error at {}. Message: {}'.format(row['name'], ex.message)
```

...?

Reply:
just using `warning` instead of `info`, check the [docs](https://docs.python.org/3.7/library/logging.html#logging.Logger.warning) and the [how to](https://docs.python.org/3/howto/logging.html#when-to-use-logging) for more details

Reply:
cool thanks!